crates/sui-json-rpc/src/read_api.rs

```diff
@@ -152,17 +152,29 @@ impl ReadApiServer for ReadApi {
             .get_executed_transaction_and_effects(digest)
             .await
             .tap_err(|err| debug!(tx_digest=?digest, "Failed to get transaction: {:?}", err))?;
-        let checkpoint = self
+        let checkpoint_seq_opt = self
             .state
             .database
             .get_transaction_checkpoint(&digest)
-            .map_err(|e| anyhow!("{e}"))?;
+            .map_err(|e| anyhow!("{e}"))?
+            .map(|(_epoch, seq)| seq);
+        let mut checkpoint_timestamp = None;
+        if let Some(checkpoint_seq) = checkpoint_seq_opt {
```

Comment:
You can assign to checkpoint_timestamp directly from map

Reply:
yeah I saw the "async closures are unstable error" can I did that, did I miss sth?

Reply:
Ah I see. Free free to merge then :)